modules/node-shared/src/main/scala/org/tessellation/node/shared/http/routes/MetagraphMessageRoutes.scala

```diff
@@ -0,0 +1,38 @@
+package org.tessellation.node.shared.http.routes
+
+import cats.effect.Async
+import cats.syntax.flatMap._
+import cats.syntax.option._
+
+import org.tessellation.node.shared.infrastructure.snapshot.services.MetagraphMessageService
+import org.tessellation.node.shared.infrastructure.snapshot.storage.MetagraphMessageStorage
+import org.tessellation.routes.internal._
+import org.tessellation.security.{Hasher, SecurityProvider}
+
+import eu.timepit.refined.auto._
+import org.http4s.HttpRoutes
+import org.http4s.dsl.Http4sDsl
+
+class MetagraphMessageRoutes[F[_]: Async: SecurityProvider: Hasher](
+  service: MetagraphMessageService[F],
+  storage: MetagraphMessageStorage[F]
+) extends Http4sDsl[F]
+    with PublicRoutes[F] {
+
+  protected val prefixPath: InternalUrlPrefix = "/currency"
+
+  protected val public: HttpRoutes[F] = HttpRoutes.of[F] {
+    case POST -> Root / "signing" / "owner" / dagAddress =>
+      service
```

Comment:
Hey, so one thing I see missing here is an invocation of a validator, I'm not certain this is implemented in the storage itself, but we need to validate upon the request whether or not it's signed by a majority of the seed list. 

Just for clarification, we can merge the signatures that are produced manually from the CLI method to produce a majority signed message.

Reply:
rewritten